### DISCUSSION TITLE
Version numbering for Actions

### DISCUSSION THREAD

**Discussion by mpdude** [community/community#25248] [Actions]
<p>When publishing an Action on the Marketplace, which version numbering scheme should I use?</p>
<p>I am not talking about SemVer, but<code>v1.2.3</code> vs. <code>1.2.3</code> (without the leading <code>v</code>). Also, does GitHub automatically apply SemVer semantics when users refer to <code>use: my/action@v1</code> and I release <code>v1.0.1</code>, <code>v1.1.0</code> etc.?</p>

**Comment by ethomson** [comment, marked as answer]
<p>We encourage you to use the leading <code>v</code>.  (For example, our checkout action is <code>actions/checkout@v1</code>.)</p>
<p>We do not <em>currently</em> apply any semantic versioning semantics, we only match a literal release tag.  However we may revisit this policy during the beta period.</p>

**Comment by mpdude** [comment]
<p>That means if I tag <code>v1</code> and want my users to benefit from bug fixes without having them update their workflow files – I‘d have to move the <code>v1</code> tag every time?</p>
<p>Also, why use <code>v1</code> if the tag does not have any meaning at all? I find that’s misleading and suggests something that’s not there yet :man_shrugging:t2:?</p>

**Comment by smac89** [comment]
<p>I found this confusing/misleading as well. After releasing an action yesterday and tagging the first version as <strong>1.0.0</strong> , I was expecting to be able to use <strong>v1</strong> to point to my <strong>1.0.0</strong> version, but was instead getting errors that the action could not be found.</p>
<p>I had read <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsuses" rel="nofollow noopener">here</a> that in order to use an action, you just need to specify the major version to recieve critical updates and I thought this meant it happens automatically. However I was wrong.</p>
<p>According to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-actions#versioning-your-action" rel="nofollow noopener">this</a> page, what they expect of the action author is that each time a minor version is released, he/she should change the tag corresponding to the major version of that release to point to the new tag, which means that the users of the action will not need to follow minor/patch releases to stay up-to-date.</p>
<blockquote>
<p>Move the major version tag (v1, v2, etc.) to point to the Git ref of the current release. For more information, see “Git basics - tagging.”</p>
</blockquote>
<p>To automate this, I created the <a href="https://github.com/marketplace/actions/actions-tagger" rel="nofollow noopener">actions-tagger</a> action which does the versioning automatically.</p>

**Comment by lukka** [comment]
<p>you should use branches (not tags) for major version, e.g. <code>v1</code> for any <code>1.x.</code>x version. Apply a fix to v1 branch (and tag with ‘v1.2.3’), then any workflow using <code>uses: action/action@v1</code> does not need to be updated. You can apply bug/security fixes to branch v1 without rewriting history (e.g. moving tags), and all those workflows would be benefit from it.</p>
<p>Tags are pretty useless, and should only be used for release notes, but unless there are specific reasons, any workflow should point to branches, e.g. <code>uses: action/action@v1</code>, not to tags e.g. <code>uses: action/action@v1.2.3</code></p>

**Comment by farshidtz** [comment]
<p>If the branch is called <code>v1</code>, workflows using the <code>action@v1</code> will checkout the head of that branch, including all the commits that aren’t tagged, the commits that are tagged with a pre-release semver or even commits that have a non semver tag. This can easily break the workflows.</p>
<p>If you insist on using a branch, it should be only kept for production and not include any pre-release code,</p>

**Comment by lukka** [comment]
<p><a class="mention" href="/farshidtz">@farshidtz</a> yes, the branch (e.g., <code>v1</code>) is reserved for production and must not contain any pre-release, all development should be on topic branches.</p>
